### ISSUE TITLE
ImportError: IPython.kernel.zmq requires pyzmq >= 13

### ISSUE THREAD

**Issue by aflyax** [yhat/rodeo#66]
``` bash
$ rodeo
Traceback (most recent call last):
  File "/home/alex/anaconda3/lib/python3.4/site-packages/IPython/utils/zmqrelated.py", line 10, in check_for_zmq
    import zmq
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/__init__.py", line 49, in <module>
    from zmq import backend
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/backend/__init__.py", line 41, in <module>
    reraise(*exc_info)
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/utils/sixcerpt.py", line 34, in reraise
    raise value
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/backend/__init__.py", line 29, in <module>
    _ns = select_backend(first)
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/backend/select.py", line 27, in select_backend
    mod = __import__(name, fromlist=public_api)
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/backend/cython/__init__.py", line 6, in <module>
    from . import (constants, error, message, context,
ImportError: libzmq.so.3: cannot open shared object file: No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alex/anaconda3/bin/rodeo", line 9, in <module>
    load_entry_point('rodeo==0.2.4', 'console_scripts', 'rodeo')()
  File "/home/alex/anaconda3/lib/python3.4/site-packages/setuptools-15.1-py3.4.egg/pkg_resources/__init__.py", line 546, in load_entry_point
  File "/home/alex/anaconda3/lib/python3.4/site-packages/setuptools-15.1-py3.4.egg/pkg_resources/__init__.py", line 2666, in load_entry_point
  File "/home/alex/anaconda3/lib/python3.4/site-packages/setuptools-15.1-py3.4.egg/pkg_resources/__init__.py", line 2339, in load
  File "/home/alex/anaconda3/lib/python3.4/site-packages/setuptools-15.1-py3.4.egg/pkg_resources/__init__.py", line 2345, in resolve
  File "/home/alex/anaconda3/lib/python3.4/site-packages/rodeo/cli.py", line 32, in <module>
    from .rodeo import main
  File "/home/alex/anaconda3/lib/python3.4/site-packages/rodeo/rodeo.py", line 1, in <module>
    from .kernel import Kernel
  File "/home/alex/anaconda3/lib/python3.4/site-packages/rodeo/kernel.py", line 5, in <module>
    from IPython.kernel import BlockingKernelClient
  File "/home/alex/anaconda3/lib/python3.4/site-packages/IPython/kernel/__init__.py", line 4, in <module>
    from . import zmq
  File "/home/alex/anaconda3/lib/python3.4/site-packages/IPython/kernel/zmq/__init__.py", line 8, in <module>
    check_for_zmq('13', 'IPython.kernel.zmq')
  File "/home/alex/anaconda3/lib/python3.4/site-packages/IPython/utils/zmqrelated.py", line 12, in check_for_zmq
    raise ImportError("%s requires pyzmq >= %s"%(required_by, minimum_version))
ImportError: IPython.kernel.zmq requires pyzmq >= 13
```

**Comment by aflyax** [comment]
Before that:

``` bash
Successfully installed markdown2-2.3.0 pyzmq-14.6.0 rodeo-0.2.4
```

**Comment by glamp** [comment]
This seems more like an IPython problem. Any chance you could try upgrading to the latest IPython?

**Comment by rdodev** [comment]
Having exact same issue:

```
raceback (most recent call last):
  File "/usr/local/bin/rodeo", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 2603, in <module>
    working_set.require(__requires__)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 666, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 565, in resolve
    raise DistributionNotFound(req)  # XXX put more info here
pkg_resources.DistributionNotFound: pyzmq>=13
```
pyzmq is indeed installed in the system and available to be imported.

**Comment by aflyax** [comment]
``` bash
$ pip install ipython --upgrade
Requirement already up-to-date: ipython in /home/alex/anaconda3/lib/python3.4/site-packages

$ pip install rodeo --upgrade
Requirement already up-to-date: rodeo in /home/alex/anaconda3/lib/python3.4/site-packages
Requirement already up-to-date: ipython>=3.0.0 in /home/alex/anaconda3/lib/python3.4/site-packages (from rodeo)
Requirement already up-to-date: Flask>=0.10.1 in /home/alex/anaconda3/lib/python3.4/site-packages (from rodeo)
Requirement already up-to-date: jedi in /home/alex/anaconda3/lib/python3.4/site-packages (from rodeo)
Requirement already up-to-date: docopt in /home/alex/anaconda3/lib/python3.4/site-packages (from rodeo)
Requirement already up-to-date: pyzmq>=13 in /home/alex/anaconda3/lib/python3.4/site-packages (from rodeo)
Requirement already up-to-date: markdown2 in /home/alex/anaconda3/lib/python3.4/site-packages (from rodeo)

$ rodeo
Traceback (most recent call last):
  File "/home/alex/anaconda3/lib/python3.4/site-packages/IPython/utils/zmqrelated.py", line 10, in check_for_zmq
    import zmq
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/__init__.py", line 49, in <module>
    from zmq import backend
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/backend/__init__.py", line 41, in <module>
    reraise(*exc_info)
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/utils/sixcerpt.py", line 34, in reraise
    raise value
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/backend/__init__.py", line 29, in <module>
    _ns = select_backend(first)
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/backend/select.py", line 27, in select_backend
    mod = __import__(name, fromlist=public_api)
  File "/home/alex/anaconda3/lib/python3.4/site-packages/zmq/backend/cython/__init__.py", line 6, in <module>
    from . import (constants, error, message, context,
ImportError: libzmq.so.3: cannot open shared object file: No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alex/anaconda3/bin/rodeo", line 9, in <module>
    load_entry_point('rodeo==0.2.4', 'console_scripts', 'rodeo')()
  File "/home/alex/anaconda3/lib/python3.4/site-packages/setuptools-15.1-py3.4.egg/pkg_resources/__init__.py", line 546, in load_entry_point
  File "/home/alex/anaconda3/lib/python3.4/site-packages/setuptools-15.1-py3.4.egg/pkg_resources/__init__.py", line 2666, in load_entry_point
  File "/home/alex/anaconda3/lib/python3.4/site-packages/setuptools-15.1-py3.4.egg/pkg_resources/__init__.py", line 2339, in load
  File "/home/alex/anaconda3/lib/python3.4/site-packages/setuptools-15.1-py3.4.egg/pkg_resources/__init__.py", line 2345, in resolve
  File "/home/alex/anaconda3/lib/python3.4/site-packages/rodeo/cli.py", line 32, in <module>
    from .rodeo import main
  File "/home/alex/anaconda3/lib/python3.4/site-packages/rodeo/rodeo.py", line 1, in <module>
    from .kernel import Kernel
  File "/home/alex/anaconda3/lib/python3.4/site-packages/rodeo/kernel.py", line 5, in <module>
    from IPython.kernel import BlockingKernelClient
  File "/home/alex/anaconda3/lib/python3.4/site-packages/IPython/kernel/__init__.py", line 4, in <module>
    from . import zmq
  File "/home/alex/anaconda3/lib/python3.4/site-packages/IPython/kernel/zmq/__init__.py", line 8, in <module>
    check_for_zmq('13', 'IPython.kernel.zmq')
  File "/home/alex/anaconda3/lib/python3.4/site-packages/IPython/utils/zmqrelated.py", line 12, in check_for_zmq
    raise ImportError("%s requires pyzmq >= %s"%(required_by, minimum_version))
ImportError: IPython.kernel.zmq requires pyzmq >= 13
```

**Comment by aflyax** [comment]
``` bash
$ pip install pyzmq --upgrade
Requirement already up-to-date: pyzmq in /home/alex/anaconda3/lib/python3.4/site-packages
```

**Comment by aflyax** [comment]
(Unfortunately that's all I know what to do at this point.)

**Comment by aflyax** [comment]
OK, I was unable to launch IPython (same exact error), so I was forced to troubleshoot better. :)

I pip-uninstalled IPython and then re-installed it and pyzmq using conda (`conda install ...`). (Somehow IPython was automatically upgraded in the past through pip when I was installing something that required the new IPython, so I just went along with it. IPython was working, so I didn't care.)

Now rodeo is running. Lookin' good. Hope this helps others.

**Comment by rdodev** [comment]
Can we please re-open this? I followed the same steps and did not. I'm still getting the same errors.

**Comment by aflyax** [comment]
Does IPython in general work?

**Comment by rdodev** [comment]
IPython works as expected and w/o issues.

**Comment by stonebig** [comment]
on Windows:
Ipython qtconsole also gets broken after playing with rodeo.
![rodeo_side_effect](https://cloud.githubusercontent.com/assets/4312421/7433378/e0c1ceb4-f030-11e4-9b57-29ce1bd77d2f.GIF)


Ipython notebook remains ok, though.

**Comment by stonebig** [comment]
Re-trying with latest pyzmq 14.6 on python 2.7.9:
- after playing with `rodeo .` + default browser, `ipython qtconsole` still works
- after playing with `rodeo .` + Qt4QtWebkit (via #30), I get the error.

**Comment by mic0331** [comment]
I have the same issue, uninstall / install with PIP did not solve the problem 

`ImportError: IPython.html requires pyzmq >= 13`

**Comment by cyrusstoller** [comment]
+1

**Comment by glamp** [comment]
this seems like an https://github.com/ipython/ipython or https://github.com/zeromq/pyzmq issue. going to close for now.

**Comment by stonebig** [comment]
do you have a specific issue number to point us to ? (on pyzmq or ipython project)

**Comment by rbohac** [comment]
This happened for me when I has installed zeromq 4.1.2

To fix I installed version 3.2.3 and it worked!
